src/components/Location/BedListing.tsx

```diff
@@ -17,6 +17,7 @@ export function BedListing({
   onBedSelect,
   onCheckStatus,
 }: BedListingProps) {
+  if (beds.length === 0) return <></>;
```

Comment:
```suggestion
  if (beds.length === 0) return null;
```

consider using null for rendering nothing, usually fragments are meant for grouping.